types/http-assert/index.d.ts

```diff
@@ -11,30 +11,30 @@
  * @param msg the message of the error, defaulting to node's text for that status code
  * @param opts custom properties to attach to the error object
  */
-declare function assert(value: any, status?: number, msg?: string, opts?: {}): void;
-declare function assert(value: any, status?: number, opts?: {}): void;
+declare function assert(value: any, status?: number, msg?: string, opts?: Record<string, any>): asserts value;
```

Comment:
I am just wondering if signature should not allow overriding `opts` type:

```
declare function assert<T = Record<string, unknown>>(value: any, status?: number, msg?: string, opts?: T): asserts value;
```

Reply:
There is no point in using a generic for one input parameter. This gives a slight performance loss when type checking and is disallowed by tslint ([no-unnecessary-generics](https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md)).

Reply:
Fair enough